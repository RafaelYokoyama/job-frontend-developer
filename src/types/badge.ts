export interface BadgeProps {
  data: { text: string; color: string }[];
}
