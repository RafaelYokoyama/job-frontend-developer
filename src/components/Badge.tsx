import React from 'react';
import { BadgeProps } from 'types/badge';

const Badge = ({ data }: BadgeProps) => {
  return (
    <div>
      {data.map((item, index) => (
        <span
          key={index}
          className={`inline-block mx-1 whitespace-nowrap rounded-[0.27rem] ${item.color} px-[0.65em] pb-[0.25em] pt-[0.35em] text-center align-baseline text-[0.75em] font-bold leading-none text-primary-700`}
        >
          {item.text}
        </span>
      ))}
    </div>
  );
};

export default Badge;
