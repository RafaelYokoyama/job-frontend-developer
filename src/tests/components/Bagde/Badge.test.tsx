import { render } from '@testing-library/react';
import Badge from 'components/Badge';
import { badgeMock } from 'tests/mocks/badge';

describe('Badge Component', () => {
  it('renders correctly', () => {
    const { container } = render(<Badge data={badgeMock} />);
    expect(container).toMatchSnapshot();
  });

  it('renders the badge component', () => {
    render(<Badge data={badgeMock} />);
  });

  it('renders the correct number of badges', () => {
    const { container } = render(<Badge data={badgeMock} />);
    const badges = container.querySelectorAll('.inline-block');
    expect(badges.length).toBe(badgeMock.length);
  });

  it('renders badges with the correct text', () => {
    const { getByText } = render(<Badge data={badgeMock} />);
    badgeMock.forEach((item) => {
      const badgeElement = getByText(item.text);
      expect(badgeElement).toBeInTheDocument();
    });
  });

  it('ensures that badges have valid data', () => {
    const { container } = render(<Badge data={badgeMock} />);
    badgeMock.forEach((item) => {
      const badgeElement = container.querySelector(`.${item.color}`);
      expect(badgeElement).toHaveTextContent(item.text);
    });
  });
});
