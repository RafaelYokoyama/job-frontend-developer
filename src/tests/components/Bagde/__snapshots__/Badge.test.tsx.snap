// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component renders correctly 1`] = `
<div>
  <div>
    <span
      class="inline-block mx-1 whitespace-nowrap rounded-[0.27rem] red px-[0.65em] pb-[0.25em] pt-[0.35em] text-center align-baseline text-[0.75em] font-bold leading-none text-primary-700"
    >
      Badge 1
    </span>
    <span
      class="inline-block mx-1 whitespace-nowrap rounded-[0.27rem] blue px-[0.65em] pb-[0.25em] pt-[0.35em] text-center align-baseline text-[0.75em] font-bold leading-none text-primary-700"
    >
      Badge 2
    </span>
    <span
      class="inline-block mx-1 whitespace-nowrap rounded-[0.27rem] green px-[0.65em] pb-[0.25em] pt-[0.35em] text-center align-baseline text-[0.75em] font-bold leading-none text-primary-700"
    >
      Badge 3
    </span>
  </div>
</div>
`;
